Source/WebCore/html/CanvasBase.cpp

```diff
@@ -361,12 +368,50 @@ RefPtr<ImageBuffer> CanvasBase::allocateImageBuffer(bool usesDisplayListDrawing,
 
 bool CanvasBase::shouldInjectNoiseBeforeReadback() const
 {
+    // Note, every early-return resulting from this check potentially leaks this state. This is a risk that we're accepting right now.
     return scriptExecutionContext() && scriptExecutionContext()->noiseInjectionHashSalt();
 }
 
-bool CanvasBase::postProcessPixelBuffer(Ref<PixelBuffer>&& pixelBuffer, bool wasLastDrawByBitMap, const HashSet<uint32_t>& suppliedColors) const
+IntRect CanvasBase::computeImageDataRect(const ImageBuffer& buffer, int width, int height, IntRect destRect, const IntSize& destOffset)
+{
+    destRect.intersect(IntRect { 0, 0, width, height });
+    destRect.move(destOffset);
+    destRect.intersect(IntRect { { }, buffer.truncatedLogicalSize() });
+    if (destRect.isEmpty())
+        return destRect;
+    IntRect sourceRect { destRect };
+    sourceRect.move(-destOffset);
+    sourceRect.intersect(IntRect { 0, 0, width, height });
+    return sourceRect;
+}
+
+void CanvasBase::postProcessDirtyCanvasBuffer() const
+{
+    if (!shouldInjectNoiseBeforeReadback())
+        return;
+
+    ImageBuffer* imageBuffer = buffer();
+    if (!imageBuffer)
+        return;
+
+    auto imageRect = !m_postProcessDirtyRect.isEmpty() ? enclosingIntRect(m_postProcessDirtyRect) : IntRect { { }, size() };
+    auto colorSpace = renderingContext() ? renderingContext()->colorSpace() : DestinationColorSpace::SRGB();
```

Comment:
Here we are postprocessing the image buffer attached to the canvas element.
as such, the image buffer itself has a color space, and we shouldn't need to consult the possible context2d for the colorspace.

Reply:
Corrected, thanks for noticing that.